Civi/RemoteParticipant/Event/RegistrationEvent.php

```diff
@@ -198,4 +198,75 @@ public function getSubmittedValue($value_name)
         return $this->submission[$value_name] ?? NULL;
     }
 
+  /**
+   * Get the event data
+   *
+   * @return array
+   *    event data
+   */
+  public function getEvent() {
+    return \CRM_Remoteevent_RemoteEvent::getRemoteEvent($this->getEventID());
+  }
+
+  /**
+   * Get the parameters of the original query
+   *
+   * @return array
+   *   parameters of the query
+   */
+  public function getQueryParameters() {
+    return $this->submission;
+  }
+
+  /**
+   * @phpstan-return array<string, mixed>
+   * @throws \CRM_Core_Exception
+   */
+  public function getPriceFieldValues(): array {
```

Comment:
This method contains some logic that might not fit into an event class that is normally just some container for data. Maybe it might be better placed in `RegistrationEventFactory`.